test/widgets/autocomplete_test.dart

```diff
@@ -48,18 +47,14 @@ Future<Finder> setupToComposeInput(WidgetTester tester, {
 
   prepareBoringImageHttpClient();
 
-  await tester.pumpWidget(
-    MaterialApp(
-      localizationsDelegates: ZulipLocalizations.localizationsDelegates,
-      supportedLocales: ZulipLocalizations.supportedLocales,
-      home: GlobalStoreWidget(
-        child: PerAccountStoreWidget(
-          accountId: eg.selfAccount.id,
-          child: MessageListPage(
-            narrow: DmNarrow(
-              allRecipientIds: [eg.selfUser.userId, eg.otherUser.userId],
-              selfUserId: eg.selfUser.userId,
-            ))))));
+  await tester.pumpWidget(const ZulipApp());
+  await tester.pump();
+  final navigator = await ZulipApp.navigator;
+  navigator.push(MessageListPage.buildRoute(accountId: eg.selfAccount.id,
+    narrow: DmNarrow(
```

Comment:
These rearrangements of existing tests are probably clearest as a separate prep commit.